src/components/Blocks/Flickr/Edit.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react';
+import { defineMessages, useIntl } from 'react-intl';
+import iconSVG from '../../../icons/flickr.svg';
+import { isValidFlickrId } from '../../../helpers';
+import { EditForm } from '../../../components';
+import { withBlockExtensions } from '@plone/volto/helpers';
+import { SidebarPortal } from '@plone/volto/components';
+
+import FlickrBlockData from './Data';
+import FlickrBlockView from './View';
+
+const messages = defineMessages({
+  editFormHeader: {
+    id: 'Embed a Flickr Gallery',
+    defaultMessage: 'Embed a Flickr Gallery',
+  },
+  editFormPlaceholder: {
+    id: 'Provide the embed code of the Flickr Gallery',
+    defaultMessage: 'Provide the embed code of the Flickr Gallery',
+  },
+  errorMessage: {
+    id: 'Please provide a valid Flickr Gallery embed code',
+    defaultMessage: 'Please provide a valid Flickr Gallery embed code',
+  },
+});
+
+const FlickrBlockEdit = (props) => {
+  const { data, onChangeBlock, block, selected } = props;
+  const [flickrId, setFlickrId] = useState(data.flickrId);
+  const [hasError, setHasError] = useState(false);
+  const intl = useIntl();
+
+  useEffect(() => {
+    if (data.flickrId !== flickrId && isValidFlickrId(data.flickrId)) {
+      setFlickrId(data.flickrId);
+    }
+  }, [data, flickrId]);
```

Comment:
@danalvrz What case is this here for? In general it's not ideal to update the component state inside useEffect. Because then the component renders once, then the effect runs, then the state changes and the component has to render again.

I don't think this is needed, because the flickrId is only updated via `updateFlickrId` which already validates it. But am I missing something?

Reply:
It wasn't, I removed it